src/pages/Main/Main.jsx

```diff
@@ -1,10 +1,145 @@
-import React from 'react';
+import { React, useState } from 'react';
+import Modal from '../../components/Modal/Modal';
+
+const INITIAL_INPUT = {
+  name: '',
+  birthday: '',
+  email: '',
+  emailOk: '',
+  password: '',
+  passwordOk: '',
+};
 
 function Main() {
+  const [showModal, setShowModal] = useState(false);
+  const [isSignUp, setIsSignUp] = useState(false);
+  const [loginInput, setLoginInput] = useState(INITIAL_INPUT);
+
+  const setValues = e => {
+    const { name, value } = e.target;
+    setLoginInput({ ...loginInput, [name]: value });
+  };
+
+  const goToLogin = () => {
+    setIsSignUp(false);
+    setShowModal(!showModal);
+  };
+
+  const goToSignUp = () => {
+    setIsSignUp(true);
+    setShowModal(!showModal);
+  };
+
+  const handleModal = () => {
+    setShowModal(!showModal);
+  };
+
+  const onClickSignUp = () => {
+    setLoginInput({
+      ...loginInput,
+      name: '',
+      birthday: '',
+      email: '',
+      emailOk: '',
+      password: '',
+      passwordOk: '',
+    });
+    setIsSignUp(!isSignUp);
```

Comment:
```suggestion
  const onClickSignUp = () => {
    setLoginInput(prev => ({
      ...prev,
      name: '',
      birthday: '',
      email: '',
      emailOk: '',
      password: '',
      passwordOk: '',
    }));
    setIsSignUp(prev => !prev);
```
함수가 외부의 값에 의존하기보다 내부의 값을 활용하는 것이 더 나은 방법입니다. 순수함수 라는 키워드로 검색 / 공부해보시면 좋을 것 같습니다!

Reply:
넵 감사합니다! 공부해보겠습니다